### PR TITLE
v3rpc: replace grpc metrics w/ go-grpc-prometheus

### PR DESCRIPTION
**Pull request by gyuho**
Discussed in https://github.com/grpc-ecosystem/go-grpc-prometheus/issues/13

Demo http://dash.etcd.io/dashboard/db/test-etcd-temp (will be shutdown once this gets merged, and existing testing cluster will be upgraded with these new metrics)

@xiang90 @heyitsanthony

### CODE REVIEW

File: integration/v3_watch_test.go [hunk 1]
```diff
@@ -1193,3 +1193,36 @@ func TestV3WatchWithPrevKV(t *testing.T) {
...
+
+	wAPI := toGRPC(clus.RandClient()).Watch
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+	wStream, err := wAPI.Watch(ctx)
+	if err != nil {
+		t.Fatalf("wAPI.Watch error: %v", err)
+	}
+
+	if err = wStream.Send(&pb.WatchRequest{RequestUnion: &pb.WatchRequest_CreateRequest{
```
Comment (heyitsanthony) [inline integration/v3_watch_test.go]: is this necessary? the metric is counting grpc streams not watches, right?

**Comment by heyitsanthony** [comment]
lgtm aside from the test; defer to @xiang90

File: integration/v3_watch_test.go (hunk 1 above)
Comment (gyuho) [inline integration/v3_watch_test.go]: You are right. Just removed. Thanks.

**Comment by xiang90** [comment]
We need to update https://github.com/coreos/etcd/blob/master/Documentation/metrics.md#grpc-requests. Probably just point to https://github.com/grpc-ecosystem/go-grpc-prometheus

**Comment by gyuho** [comment]
@xiang90 Just updated `metrics.md` to just point to https://github.com/grpc-ecosystem/go-grpc-prometheus

**Comment by xiang90** [comment]
LGTM

**Comment by gyuho** [comment]
merging. will handle grpc-proxy handler in a separate PR

**Pull request merged**